src/models/invites-ignorer.ts

```diff
@@ -227,21 +242,30 @@ export class IgnoredInvites {
             const room = this.client.getRoom(target);
             if (room) {
                 return room;
-            } else {
-                target = null;
             }
         }
-        // We need to create our own policy room for ignoring invites.
-        target = (await this.client.createRoom({
-            name: "Individual Policy Room",
-            preset: Preset.PrivateChat,
-        })).room_id;
-        await this.withIgnoreInvitesPolicies(ignoreInvitesPolicies => {
-            ignoreInvitesPolicies.target = target;
-        });
+        try {
+            // We need to create our own policy room for ignoring invites.
+            await this._getOrCreateTargetRoomLock.acquireAsync();
+            target = (await this.client.createRoom({
+                name: "Individual Policy Room",
+                preset: Preset.PrivateChat,
+            })).room_id;
```

Comment:
Does this need to lock? We could instead return the same promise, no?

Reply:
Which promise? That one?

```ts
this.client.createRoom({
                name: "Individual Policy Room",
                preset: Preset.PrivateChat,
            })
```
We could probably store it and use it as a makeshift lock, but it feels more fragile/harder to read. Since we don't expect that this block will be performance-critical (in most runs, it should be executed either never or once), I'd favour the most readable solution, whichever it is.

Reply:
We have that pattern in other places - https://github.com/matrix-org/matrix-js-sdk/blob/8aee884d03d79934e8944b05c22643c1bb480a64/src/models/room.ts#L409-L427 

I wouldn't suggest just storing the promise of the createRoom call, but rather this whole method by wrapping it.

Reply:
Yeah, that would work but it feels a bit more brittle in presence of exception.

Your call.

Reply:
Done.
As always, documenting concurrency is an interesting pastime :)